Source/Icebreaker/Services/MatchingService.cs

```diff
@@ -185,12 +197,51 @@ private string GetChannelUserObjectId(ChannelAccount account)
             return JObject.FromObject(account).ToObject<TeamsChannelAccount>()?.AadObjectId;
         }
 
+        /// <summary>
+        /// Parse through a list of pairing information.
+        /// </summary>
+        /// <param name="pairHistory">List of pairing information.</param>
+        /// <returns>Returns pairs from the most recent pairing phase. This can be changed to include
+        /// pairings from less recent pairing phases by changing the `latestIteration` threshold.</returns>
+        private Tuple<Dictionary<string, HashSet<string>>, int> ParsePairHistory(IList<PairInfo> pairHistory)
+        {
+            var pastPairs = new Dictionary<string, HashSet<string>>();
+            int latestIteration = 0;
+
+            foreach (var pair in pairHistory)
+            {
+                latestIteration = (latestIteration >= pair.Iteration) ? latestIteration : pair.Iteration;
```

Comment:
NIT: I would just use `Math.Max` here to be explicit on what is being done. 

Also as a trick you can do this entire loop with LINQ
```cs
int latestIteration = pairHistory.Select(pair => pair.Iteration).Max();
```